src/main/java/seedu/kitchenhelper/command/ListCommand.java

```diff
@@ -3,21 +3,82 @@
 import seedu.kitchenhelper.exception.KitchenHelperException;
 import seedu.kitchenhelper.object.Chore;
 import seedu.kitchenhelper.object.Recipe;
+import seedu.kitchenhelper.object.ingredient.Vegetable;
+import seedu.kitchenhelper.object.ingredient.Staple;
+import seedu.kitchenhelper.object.ingredient.Miscellaneous;
+import seedu.kitchenhelper.object.ingredient.Dairy;
+import seedu.kitchenhelper.object.ingredient.Drink;
+import seedu.kitchenhelper.object.ingredient.Fruit;
+import seedu.kitchenhelper.object.ingredient.Meat;
 import seedu.kitchenhelper.object.ingredient.Ingredient;
-
 import java.util.ArrayList;
+import java.util.HashMap;
 
 public class ListCommand extends Command {
     
     public static final String COMMAND_WORD = "list";
-    
-    public void listIngredients(String attributes) {
-    
+    public static final String COMMAND_FORMAT = "list ingredient";
+
+    public void setListParams(HashMap<String, String> attributes) {
+        actionType = COMMAND_WORD;
+        setTypeOfObject(attributes.get("type"));
+        if (attributes.size() == 2) {
+            objectVariables = attributes.get("item");
+        }
     }
-    
-    public void listRecipe(String attributes) {
-    
+
+    public void setTypeOfObject(String object) {
+        String[] attributes = object.split("\\s");
+        if (attributes[0].equalsIgnoreCase("recipe")) {
+            objectType = "recipe";
+        } else if (attributes[0].equalsIgnoreCase("ingredient")) {
+            objectType = "ingredient";
+        } else if (attributes[0].equalsIgnoreCase("chore")) {
+            objectType = "chore";
+        }
+    }
+
+    public String listIngredients(ArrayList<Ingredient> ingredientList) {
+        String result = "Here is the list of Ingredients in Inventory:\n"
+                + "Format : Ingredient Name | Portion | Price | Expiry\n";
```

Comment:
It will be more consistent to use quantity instead of portion, and would it be better to include category as well?